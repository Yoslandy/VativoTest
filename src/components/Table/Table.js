import React from "react"
import PropTypes from "prop-types"

export default function Table(props) {
    // eslint-disable-next-line react/prop-types
    const { children, className, color } = props
    const getBgColor = () => {
        if (color === "dark")
            return "items-center w-full border-collapse bg-sky-900 text-white"
        return "items-center w-full border-collapse"
    }
    return <table className={`${getBgColor()} ${className}`}>{children}</table>
}

Table.propTypes = {
    className: PropTypes.string,
    color: PropTypes.oneOf(["ligth", "dark"]),
}

Table.defaultProps = {
    className: "",
    color: "ligth",
}
