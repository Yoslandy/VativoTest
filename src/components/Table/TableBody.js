import React from "react"
import PropTypes from "prop-types"

export default function TableBody(props) {
    // eslint-disable-next-line react/prop-types
    const { children, className } = props
    return <tbody className={className}>{children}</tbody>
}

TableBody.propTypes = {
    className: PropTypes.string,
}

TableBody.defaultProps = {
    className: "",
}
