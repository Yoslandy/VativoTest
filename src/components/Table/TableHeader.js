import React from "react"
import PropTypes from "prop-types"

export default function TableHeader(props) {
    // eslint-disable-next-line react/prop-types
    const { children, className } = props
    return <thead className={className}>{children}</thead>
}

TableHeader.propTypes = {
    className: PropTypes.string,
}

TableHeader.defaultProps = {
    className: "",
}
