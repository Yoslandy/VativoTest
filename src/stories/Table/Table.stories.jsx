import React from "react"

import "@fortawesome/fontawesome-free/css/all.min.css"
import "../../assets/styles/build.css"

import Table from "./Table"

export default {
    title: "Example/Table",
    component: Table,
    argTypes: {},
}

function Template(args) {
    return <Table {...args} />
}

export const Default = Template.bind({})

export const Dark = Template.bind({})
Dark.args = {
    color: "dark",
}
