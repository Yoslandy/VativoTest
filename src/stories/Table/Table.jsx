import React from "react"
import PropTypes from "prop-types"

import Table from "../../components/Table/Table"
import TableBody from "../../components/Table/TableBody"
import TableHeader from "../../components/Table/TableHeader"

export default function CustomTable({ className, color }) {
    const headers = ["Firstname", "Lastname", "Native", "User Preferred"]
    const data = [
        {
            first: "Spike",
            last: "Spiegel",
            native: "スパイク・スピーゲル",
            userPreferred: "Spike Spiegel",
        },
        {
            first: "Faye",
            last: "Valentine",
            native: "フェイ・バレンタイン",
            userPreferred: "Faye Valentine",
        },
        {
            first: "Jet",
            last: "Black",
            native: "ジェット・ブラック",
            userPreferred: "Jet Black",
        },
        {
            first: "Spike",
            last: "Spiegel",
            native: "スパイク・スピーゲル",
            userPreferred: "Spike Spiegel",
        },
    ]
    return (
        <Table className={className} color={color}>
            <TableHeader>
                <tr>
                    {headers.map(header => (
                        <th
                            className={`px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left ${
                                color === "light"
                                    ? "bg-slate-50 text-slate-500 border-slate-100"
                                    : "bg-sky-800 text-sky-300 border-sky-700"
                            }`}
                        >
                            {header}
                        </th>
                    ))}
                </tr>
            </TableHeader>
            <TableBody>
                {data.map(d => (
                    <tr>
                        <th className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4 text-left">
                            {d.first}
                        </th>
                        <td className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4">
                            {d.last}
                        </td>
                        <td className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4">
                            {d.native}
                        </td>
                        <td className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4">
                            <i className="fas fa-arrow-up text-emerald-500 mr-4" />
                            {d.userPreferred}
                        </td>
                    </tr>
                ))}
            </TableBody>
        </Table>
    )
}

CustomTable.propTypes = {
    className: PropTypes.string,
    color: PropTypes.string,
}

CustomTable.defaultProps = {
    className: "",
    color: "light",
}
