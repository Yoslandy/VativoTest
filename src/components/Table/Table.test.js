import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import Table from "./Table"
import TableBody from "./TableBody"
import TableHeader from "./TableHeader"

test("Renders Table form correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Table className="items-center w-full bg-transparent border-collapse">
                <TableHeader>
                    <tr>
                        <th className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left">
                            Firstname
                        </th>
                        <th className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left">
                            Lastname
                        </th>
                        <th className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left">
                            Native
                        </th>
                        <th className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left">
                            User Preferred
                        </th>
                    </tr>
                </TableHeader>
                <TableBody>
                    <tr>
                        <th className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4 text-left">
                            Spike
                        </th>
                        <td className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4">
                            Spiegel
                        </td>
                        <td className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4">
                            スパイク・スピーゲル
                        </td>
                        <td className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4">
                            <i className="fas fa-arrow-up text-emerald-500 mr-4" />
                            Spike Spiegel
                        </td>
                    </tr>
                </TableBody>
            </Table>
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
