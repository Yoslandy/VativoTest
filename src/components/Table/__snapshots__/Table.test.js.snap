// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Table form correctly 1`] = `
<table
  className="items-center w-full border-collapse items-center w-full bg-transparent border-collapse"
>
  <thead
    className=""
  >
    <tr>
      <th
        className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left"
      >
        Firstname
      </th>
      <th
        className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left"
      >
        Lastname
      </th>
      <th
        className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left"
      >
        Native
      </th>
      <th
        className="px-6 bg-slate-50 text-slate-500 align-middle border border-solid border-slate-100 py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left"
      >
        User Preferred
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr>
      <th
        className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4 text-left"
      >
        Spike
      </th>
      <td
        className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
      >
        Spiegel
      </td>
      <td
        className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
      >
        スパイク・スピーゲル
      </td>
      <td
        className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-nowrap p-4"
      >
        <i
          className="fas fa-arrow-up text-emerald-500 mr-4"
        />
        Spike Spiegel
      </td>
    </tr>
  </tbody>
</table>
`;
